npm/private/utils.bzl

```diff
@@ -434,29 +491,27 @@ def _convert_v9_packages(packages, snapshots):
             fail(msg)
 
         # the raw name + version are the key, not including peerDeps+patch
-        version_index = static_key.find("@", 1)
+        version_index = static_key.index("@", 1)
         name = static_key[:version_index]
-        friendly_version = static_key[version_index + 1:]
         package_key = _convert_pnpm_v6_v9_version_peer_dep(package_key)
 
         # Extract the version including peerDeps+patch from the key
-        version = package_key[package_key.find("@", 1) + 1:]
+        version = package_key[package_key.index("@", 1) + 1:]
+
+        # package_data can have the resolved "version" for things like https:// deps
+        friendly_version = package_data["version"] if "version" in package_data else static_key[version_index + 1:]
 
         # Convert the package_key to a rules_js compatible format
         package_key = _to_package_key(name, version)
 
-        # package_data can have the resolved "version" for things like https:// deps
-        if "version" in package_data:
-            friendly_version = package_data["version"]
-
         package_info = _new_package_info(
-            id = package_data.get("id", None),
+            id = package_data.get("id", None),  # TODO: does v9 have "id"?
             name = name,
             version = version,
             friendly_version = friendly_version,
             dependencies = _convert_pnpm_v9_package_dependency_map(package_snapshot.get("dependencies", {})),
             optional_dependencies = _convert_pnpm_v9_package_dependency_map(package_snapshot.get("optionalDependencies", {})),
-            peer_dependencies = _convert_pnpm_v9_package_dependency_map(package_data.get("peerDependencies", {})),
+            peer_dependencies = package_data.get("peerDependencies", {}),
```

Comment:
Peer data is from the static `packages` data, not the `snapshots` where all the peer data has been mixed in so the callback isn't needed.